src/lib.rs

```diff
@@ -309,24 +314,27 @@ impl<F: io::Write> io::Write for File<F> {
 #[derive(Debug, Clone, Copy)]
 pub struct DelimCodec<D>(pub D);
 
-impl<D: Into<u8> + Copy> tokio_core::io::Codec for DelimCodec<D> {
-    type In = Vec<u8>;
-    type Out = Vec<u8>;
+impl<D: Into<u8> + Copy> codec::Decoder for DelimCodec<D> {
+    type Item = Vec<u8>;
+    type Error = io::Error;
 
-    fn decode(&mut self, buf: &mut tokio_core::io::EasyBuf)
-              -> io::Result<Option<Self::In>> {
+    fn decode(&mut self, buf: &mut BytesMut) -> io::Result<Option<Self::Item>> {
         Ok(buf.as_ref().iter().position(|b| *b == self.0.into())
-           .map(|n| buf.drain_to(n + 1).as_ref().to_vec()))
+           .map(|n| buf.split_to(n + 1).as_ref().to_vec()))
     }
 
-    fn decode_eof(&mut self, buf: &mut tokio_core::io::EasyBuf)
-                  -> io::Result<Self::In> {
-        Ok(buf.split_off(0).as_ref().to_vec())
+    fn decode_eof(&mut self, buf: &mut BytesMut) -> io::Result<Option<Self::Item>> {
+        Ok(Some(buf.split_off(0).as_ref().to_vec()))
```

Comment:
Oops, sorry.  Looks I missed something important.

It seems the interface has [changed](https://docs.rs/tokio-io/0.1/tokio_io/codec/trait.Decoder.html#method.decode_eof) from `io::Result<Self::In>` to `io::Result<Option<Self::Item>>`.  If the `buf.len() == 0`, then it must return `Ok(None)` otherwise this part would loop forever.